ddl/ddl_api.go

```diff
@@ -3091,7 +3093,7 @@ func (d *ddl) AlterTableCharsetAndCollate(ctx sessionctx.Context, ident ast.Iden
 		SchemaName: schema.Name.L,
 		Type:       model.ActionModifyTableCharsetAndCollate,
 		BinlogInfo: &model.HistoryInfo{},
-		Args:       []interface{}{toCharset, toCollate},
+		Args:       []interface{}{toCharset, toCollate, needsOverwriteCols},
```

Comment:
Will this cause compatible problem? @zimulala PTAL

Reply:
I think it is fine. When rolling upgrade, it is equivalent to not solving this problem.